mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/extractor/mapping/programmatic/ContainerExtractorPath.java

```diff
@@ -36,11 +36,18 @@
  */
 public class ContainerExtractorPath {
 
+	public enum DefaultExtractorsApplicability {
+		ANY, SINGLE_VALUED, NO;
+	}
+
 	private static final ContainerExtractorPath DEFAULT = new ContainerExtractorPath(
-			true, Collections.emptyList()
+			DefaultExtractorsApplicability.ANY, Collections.emptyList()
+	);
+	private static final ContainerExtractorPath DEFAULT_SINGLE_VALUED = new ContainerExtractorPath(
+			DefaultExtractorsApplicability.SINGLE_VALUED, Collections.emptyList()
 	);
 	private static final ContainerExtractorPath NONE = new ContainerExtractorPath(
-			false, Collections.emptyList()
+			DefaultExtractorsApplicability.NO, Collections.emptyList()
 	);
```

Comment:
Does it make sense to have it like this? It's not what you were suggesting, but it looked less intrusive in terms of adding the boolean parameter everywhere ... 🙈

Reply:
I was hoping my suggestion would only force you to add the boolean (or enum, whatever) from `org.hibernate.search.mapper.pojo.mapping.building.spi.PojoIndexMappingCollectorPropertyNode#value` down to `org.hibernate.search.mapper.pojo.extractor.impl.ContainerExtractorBinder#bindPath`, and that it wouldn't have too many side-effects... I guess I was wrong?

Changing `ContainerExtractorPath` is a possibility, but it has ramifications as well, this time on APIs... We'd need to start thinking about what it means for users to use `defaultSingleValued()` extractor paths elsewhere...

And in fact I suspect that having multiple intepretations of what is the "default" will lead to problems with `org.hibernate.search.mapper.pojo.extractor.impl.ContainerExtractorBinder#isDefaultExtractorPath`...

Maybe it would be better to stick to what you initially did for now, and introduce support for "default" extractors once we have the time to design a future-proof solution that takes into account all the possible corner cases?

Reply:
yeah, it seems like that would be a safer option ... see the second commit I've just added... as soon as I update `ContainerExtractorBinder#bindPath` and I haven't even changed the `ContainerExtractorBinder#tryBindPath` 😔

Reply:
I'll give you my perfectly objective, non-biased and very nuanced opinion: `indexNullAs` is shit ;D

I mean, what does it even mean to index a missing value as `0` or `"somestring"` or `[0,0,0]`? Why would you want missing values to be considered close to `[0,0,0]`? Can one even come up with a vector (`[0,0,0]` or other) that properly represents a "missing value" in the vector space of your AI model? I doubt it.

`Optional<float[]>` would just be a more fancy way to say "I may not have data for this". E.g. "this book doesn't have a cover", or "I haven't turned the cover of this book into a vector yet".
So, yeah, maybe not a mainstream use case. But a relevant one nonetheless. 

Anyway. Let's go with your first solution, it has the advantage of being clear and straightforward, allows people to map `Option<float[]>` if they want to, and opens the door to supporting default extractors in the future, with some more changes.

Thanks for investigating!